File: packages/usb_capture/lib/src/lan_http_url.dart
class LanHttpUrl {
  static String display({required String ipv4, required int port}) {
    return 'http://$ipv4:$port/';
  }

  static String? pickIpv4(Iterable<String> addresses) {
    for (final raw in addresses) {
      final host = raw.trim();
      if (host.isEmpty || host == '127.0.0.1' || host == '0.0.0.0') continue;
      if (host.contains(':')) continue;
      final parts = host.split('.');
      if (parts.length != 4) continue;
      if (parts.every((p) => int.tryParse(p) != null)) return host;
    }
    return null;
  }
}

File: packages/usb_capture/android/src/main/kotlin/com/usbcamera/capture/usb_capture/LanHttpUrl.kt
package com.usbcamera.capture.usb_capture

internal object LanHttpUrl {
    fun display(ipv4: String, port: Int): String {
        return "http://$ipv4:$port/"
    }

    fun pickIpv4(addresses: Iterable<String>): String? {
        for (raw in addresses) {
            val host = raw.trim()
            if (host.isEmpty() || host == "127.0.0.1" || host == "0.0.0.0") continue
            if (host.contains(':')) continue
            val parts = host.split('.')
            if (parts.size != 4) continue
            if (parts.all { it.toIntOrNull() != null }) return host
        }
        return null
    }
}
